packages/core/src/Provider/providerUtil.ts

```diff
@@ -15,6 +15,7 @@ export const withProviderErrors = <T extends {}>(providerImplementation: T, toPr
   }, {} as any) as T;
 
 const tryParseBigIntKey = (key: string) => {
+  if (key.startsWith('0x')) return key.slice(2);
```

Comment:
Better to do a separate commit for this and adding an inline comment to explain why; if I remember correctly to support CIP-005 v2